app/service/planning_svc.py

```diff
@@ -4,42 +4,49 @@
 from base64 import b64decode
 
 from app.objects.c_link import Link
-from app.utility.base_service import BaseService
+from app.utility.base_planning_svc import BasePlanningService
 from app.utility.rule import RuleSet
 
 
-class PlanningService(BaseService):
+class PlanningService(BasePlanningService):
 
     def __init__(self):
         self.log = self.add_service('planning_svc', self)
 
-    async def select_links(self, operation, agent, phase):
+    async def get_links(self, operation, agent, phase=None, trim=True):
         """
-        For an operation, phase and agent combination, determine which (potential) links can be executed
+        For an operation, phase and agent combination, create links (that can be executed)
         :param operation:
         :param agent:
         :param phase:
+        :param trim: call trim_links() on list of links before returning
         :return: a list of links
         """
         if (not agent.trusted) and (not operation.allow_untrusted):
             self.log.debug('Agent %s untrusted: no link created' % agent.paw)
             return []
-        phase_abilities = [i for p, v in operation.adversary.phases.items() if p <= phase for i in v]
-        link_status = await self._default_link_status(operation)
 
+        if phase:
+            abilities = [i for p, v in operation.adversary.phases.items() if p <= phase for i in v]
+        else:
+            abilities = [i for p, v in operation.adversary.phases.items() for i in v]
+    
+        link_status = await self._default_link_status(operation)
         links = []
-        for a in await agent.capabilities(phase_abilities):
+        for a in await agent.capabilities(abilities):
             links.append(
-                Link(operation=operation.name, command=a.test, paw=agent.paw, score=0, ability=a,
+                 Link(operation=operation.name, command=a.test, paw=agent.paw, score=0, ability=a,
                      status=link_status, jitter=self.jitter(operation.jitter))
             )
-        ability_requirements = {ab.unique: ab.requirements for ab in phase_abilities}
-        links[:] = await self._trim_links(operation, links, agent, ability_requirements)
+        if trim:
+            ability_requirements = {ab.unique: ab.requirements for ab in abilities}
+            links[:] = await self.trim_links(operation, links, agent, ability_requirements)
+
         return await self._sort_links(links)
 
-    async def select_cleanup_links(self, operation, agent):
+    async def get_cleanup_links(self, operation, agent):
```

Comment:
Same for above, make the `agent` field optional and return all cleanup links by default.  Otherwise return cleanup links for a specific agent if it is passed.